sktime/transformations/series/temporian.py

```diff
@@ -0,0 +1,115 @@
+            "ianspektor"
+        ],  # TODO: add other Temporian devs who contributed to this
+        "maintainers": [
+            "ianspektor"
+        ],  # TODO: add other Temporian devs who will maintain this
+        "scitype:transform-input": "Series",
+        "scitype:transform-output": "Series",
+        "scitype:instancewise": True,
+        "scitype:transform-labels": "None",
+        # TODO: extend to pd-multiindex and pd_multiindex_hier
+        "X_inner_mtype": "pd.DataFrame",
+        "fit_is_empty": True,
+        "capability:inverse_transform": False,
+        "capability:unequal_length": True,
+        "handles-missing-data": False,
+        "python_dependencies": ["temporian"],
+        "python_version": "<3.12",
+    }
+
+    def __init__(self, function):
+        self.function = function
+
+        super().__init__()
+
+        # TODO: ensure function receives a single EventSet/param with inspect module?
+        #       or is failing in runtime OK?
+        # TODO: @tp.compile the function
+
+    def _transform(self, X, y=None):
+        """Transform X and return a transformed version.
+
+        private _transform containing core logic, called from transform
+
+        Parameters
+        ----------
+        X : pd.DataFrame
+            if self.get_tag("univariate-only")==True:
+                guaranteed to have a single column
+            if self.get_tag("univariate-only")==False: no restrictions apply
+        y : None, present only for interface compatibility
+
+        Returns
+        -------
+        transformed version of X
+        """
+        import temporian as tp
+
+        timestamps_col = X.index.name
+        X_noindex = X.reset_index(drop=False)
+        evset = tp.from_pandas(X_noindex, timestamps=timestamps_col)
+
+        # TODO: check function returns a single EventSet
+        res = self.function(evset)
+
+        res = tp.to_pandas(res).rename(columns={"timestamp": timestamps_col})
+        res = res.set_index(timestamps_col).to_period()
+        return res
+
+    @classmethod
+    def get_test_params(cls, parameter_set="default"):
+        """Return testing parameter settings for the estimator.
+
+        Parameters
+        ----------
+        parameter_set : str, default="default"
+            Name of the set of test parameters to return, for use in tests. If no
+            special parameters are defined for a value, will return `"default"` set.
+            There are currently no reserved values for transformers.
+
+        Returns
+        -------
+        params : dict or list of dict, default = {}
+            Parameters to create testing instances of the class
+            Each dict are parameters to construct an "interesting" test instance, i.e.,
+            `MyClass(**params)` or `MyClass(**params[i])` creates a valid test instance.
+            `create_test_instance` uses the first (or only) dictionary in `params`
+        """
+        import temporian as tp
+
+        return {
```

Comment:
optimally, we have at least two valid examples, that enables additional tests like parameter switch.